javav2/usecases/workflow_multiple_channels/Readme.md

```diff
@@ -0,0 +1,1122 @@
+       import javax.xml.transform.Transformer;
+       import javax.xml.transform.TransformerException;
+       import javax.xml.transform.TransformerFactory;
+       import javax.xml.transform.dom.DOMSource;
+       import javax.xml.transform.stream.StreamResult;
+       import java.io.*;
+       import java.sql.*;
+       import java.util.ArrayList;
+       import java.util.List;
+
+       public class RDSGetStudents {
+
+        public String getStudentsRDS(String date ) {
+
+        Connection c = null;
+        String query = "";
+
+        try {
+
+            c = ConnectionHelper.getConnection();
+            ResultSet rs = null;
+
+            // Use prepared statements.
+            PreparedStatement pstmt = null;
+            query = "Select first, phone, mobile, email FROM students where date = '" +date +"'";
+            pstmt = c.prepareStatement(query);
+            rs = pstmt.executeQuery();
+
+            List<Student> studentList = new ArrayList<>();
+            while (rs.next()) {
+
+                Student student = new Student();
+
+                String name = rs.getString(1);
+                String phone = rs.getString(2);
+                String mobile = rs.getString(3);
+                String email = rs.getString(4);
+
+                student.setFirstName(name);
+                student.setMobileNumber(mobile);
+                student.setPhoneNunber(phone);
+                student.setEmail(email);
+
+                // Push the Student object to the list.
+                studentList.add(student);
+            }
+
+            return convertToString(toXml(studentList));
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            ConnectionHelper.close(c);
+        }
+        return null;
+    }
+
+    private String convertToString(Document xml) {
+        try {
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            StreamResult result = new StreamResult(new StringWriter());
+            DOMSource source = new DOMSource(xml);
+            transformer.transform(source, result);
+            return result.getWriter().toString();
+
+        } catch(TransformerException ex) {
+            ex.printStackTrace();
+        }
+        return null;
+    }
+
+
+    // Convert the list to XML.
+    private Document toXml(List<Student> itemList) {
+
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document doc = builder.newDocument();
+
+            // Start building the XML.
+            Element root = doc.createElement( "Students" );
+            doc.appendChild( root );
+
+            // Get the elements from the collection.
+            int studentCount = itemList.size();
+
+            // Iterate through the collection.
+            for ( int index=0; index < studentCount; index++) {
```

Comment:
Use for-each: `for(Student myStudent: itemList)`

Reply:
fixed